src/worker.cpp

```diff
@@ -1185,4 +1183,305 @@ void Worker::collectFrusFromJson()
+    {
+        fillVPDMap(l_primaryHwPath, l_parsedPrimaryVpdMap);
+    }
+
+    types::VPDMapVariant l_parsedSecVpdMap;
+    if (!l_secondaryHwPath.empty())
+    {
+        fillVPDMap(l_secondaryHwPath, l_parsedSecVpdMap);
+    }
+
+    if (!l_primaryHwPath.empty() && !l_secondaryHwPath.empty() &&
+        l_parsedPrimaryVpdMap.index() != l_parsedSecVpdMap.index())
+    {
+        logging::logMessage(
+            "Primary and secondary VPD types are not matching.");
+        return;
+    }
+
+    // Implement backup and restore for IPZ type VPD
+    if (auto l_backupMapPtr = std::get_if<types::IpzBackupMap>(&l_backupMap))
+    {
+        types::IPZVpdMap* l_parsedVpdPtr = nullptr;
+        if (!(l_parsedVpdPtr = std::get_if<types::IPZVpdMap>(&io_parsedVpdMap)))
+        {
+            if (std::holds_alternative<std::monostate>(io_parsedVpdMap))
+            {
+                l_parsedVpdPtr = new types::IPZVpdMap();
+            }
+            else
+            {
+                logging::logMessage("Parsed VPD is not of IPZ type.");
+                return;
+            }
+        }
+
+        types::IPZVpdMap* l_primaryVpdPtr;
+        if (!(l_primaryVpdPtr =
+                  std::get_if<types::IPZVpdMap>(&l_parsedPrimaryVpdMap)))
+        {
+            if (std::holds_alternative<std::monostate>(l_parsedPrimaryVpdMap))
+                l_primaryVpdPtr = new types::IPZVpdMap();
+            else
+            {
+                logging::logMessage("Primary VPD is not of IPZ type.");
+                return;
+            }
+        }
+
+        types::IPZVpdMap* l_secVpdPtr;
+        if (!(l_secVpdPtr = std::get_if<types::IPZVpdMap>(&l_parsedSecVpdMap)))
+        {
+            if (std::holds_alternative<std::monostate>(l_parsedSecVpdMap))
+                l_secVpdPtr = new types::IPZVpdMap();
+            else
+            {
+                logging::logMessage("Secondary VPD is not of IPZ type.");
+                return;
+            }
+        }
+
+        backupAndRestoreIpzVpd(*l_parsedVpdPtr, *l_backupMapPtr,
+                               *l_primaryVpdPtr, *l_secVpdPtr);
+    }
+    else if (std::holds_alternative<types::KeywordVpdMap>(
+                 l_parsedPrimaryVpdMap) ||
+             std::holds_alternative<types::KeywordVpdMap>(l_parsedSecVpdMap))
+    {
+        // To Do, handle implementation for Keyword type VPD
+    }
+}
+
+void Worker::backupAndRestoreIpzVpd(types::IPZVpdMap& io_parsedVpdMap,
+                                    const types::IpzBackupMap& i_backupInfoMap,
+                                    const types::IPZVpdMap& i_primaryVpdMap,
+                                    const types::IPZVpdMap& i_secondaryVpdMap)
+{
+    bool l_isParsedMapEmpty = io_parsedVpdMap.empty();
+    const std::string& l_primaryHwPath = getBackupHwPath("primary");
+    const std::string& l_secondaryHwPath = getBackupHwPath("secondary");
+    const auto& l_invPathForPrimary = getInvPathForBackupPath("primary");
```

Comment:
First the hw path is fetched from API "getBackupHwPath" and then again the same thing is done inside "getInvPathForBackupPath"? Why?